src/main/java/slimeknights/tconstruct/plugin/crt/managers/CastingBasinManager.java

```diff
@@ -0,0 +1,51 @@
+package slimeknights.tconstruct.plugin.crt.managers;
+
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import net.minecraft.item.Item;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.ResourceLocation;
+import org.openzen.zencode.java.ZenCodeType;
+import slimeknights.mantle.recipe.FluidIngredient;
+import slimeknights.mantle.recipe.ItemOutput;
+import slimeknights.tconstruct.library.materials.MaterialId;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.library.recipe.casting.ICastingRecipe;
+import slimeknights.tconstruct.library.recipe.casting.ItemCastingRecipe;
+import slimeknights.tconstruct.library.recipe.casting.container.ContainerFillingRecipe;
+import slimeknights.tconstruct.library.recipe.casting.material.CompositeCastingRecipe;
+import slimeknights.tconstruct.library.recipe.casting.material.MaterialCastingRecipe;
+import slimeknights.tconstruct.library.tinkering.IMaterialItem;
+import slimeknights.tconstruct.plugin.crt.managers.base.ICastingManager;
+
+@ZenRegister
+@ZenCodeType.Name("mods.tconstruct.CastingBasin")
+public class CastingBasinManager implements ICastingManager {
+  
+  @Override
+  public ItemCastingRecipe makeItemCastingRecipe(ResourceLocation id, String group, Ingredient cast, FluidIngredient fluid, ItemOutput result, int coolingTime, boolean consumed, boolean switchSlots) {
+    return new ItemCastingRecipe.Basin(id, "", cast, fluid, result, coolingTime, consumed, switchSlots);
```

Comment:
You have a group parameter, but then pass in an empty string for group here. Few other places in this file